File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int left = 0, right = nums.length - 1, current = nums.length - 1;
        int[] newArray = new int[nums.length];
        while (left <= right) {
            if (Math.abs(nums[right]) >= Math.abs(nums[left])) {
                newArray[current--] = nums[right] * nums[right];
                right--;
            } else {
                newArray[current--] = nums[left] * nums[left];
                left++;

            }
        }
        return newArray;
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left, right = 0, len(nums) - 1

        res = [0] * len(nums)
        idx = len(nums) - 1

        while left <= right:
            rightSquared = nums[right] * nums[right]
            leftSquared = nums[left] * nums[left]
            if rightSquared >= leftSquared:
                res[idx] = rightSquared
                right -= 1
            else:
                res[idx] = leftSquared
                left += 1
            
            idx -= 1

        return res